pymodbus/pdu/decoders.py

```diff
@@ -102,7 +102,7 @@ def decode(self, frame: bytes) -> base.ModbusPDU | None:
         """Decode a frame."""
         try:
             if (function_code := int(frame[0])) > 0x80:
-                pdu_exp = base.ExceptionResponse(function_code & 0x7F)
+                pdu_exp = base.ExceptionResponse(function_code & 0x7F, int(frame[1]))
```

Comment:
please do not convert to int, if a conversion is needed it should be done where needed.